jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/JvmIntegrationTest.java

```diff
@@ -44,30 +44,38 @@ protected MetricsVerifier createMetricsVerifier() {
             "PS Survivor Space");
     List<String> gcCollectionLabels = Arrays.asList("PS MarkSweep", "PS Scavenge");
 
-    return MetricsVerifier.create()
-        .assertGauge("jvm.classes.loaded", "number of loaded classes", "1")
-        .assertTypedCounter(
-            "jvm.gc.collections.count",
-            "total number of collections that have occurred",
-            "1",
-            gcCollectionLabels)
-        .assertTypedCounter(
-            "jvm.gc.collections.elapsed",
-            "the approximate accumulated collection elapsed time in milliseconds",
-            "ms",
-            gcCollectionLabels)
-        .assertGauge("jvm.memory.heap.committed", "current heap usage", "by")
-        .assertGauge("jvm.memory.heap.init", "current heap usage", "by")
-        .assertGauge("jvm.memory.heap.max", "current heap usage", "by")
-        .assertGauge("jvm.memory.heap.used", "current heap usage", "by")
-        .assertGauge("jvm.memory.nonheap.committed", "current non-heap usage", "by")
-        .assertGauge("jvm.memory.nonheap.init", "current non-heap usage", "by")
-        .assertGauge("jvm.memory.nonheap.max", "current non-heap usage", "by")
-        .assertGauge("jvm.memory.nonheap.used", "current non-heap usage", "by")
-        .assertTypedGauge("jvm.memory.pool.committed", "current memory pool usage", "by", gcLabels)
-        .assertTypedGauge("jvm.memory.pool.init", "current memory pool usage", "by", gcLabels)
-        .assertTypedGauge("jvm.memory.pool.max", "current memory pool usage", "by", gcLabels)
-        .assertTypedGauge("jvm.memory.pool.used", "current memory pool usage", "by", gcLabels)
-        .assertGauge("jvm.threads.count", "number of threads", "1");
+    MetricsVerifier metricsVerifier =
+        MetricsVerifier.create()
+            .assertGauge("jvm.classes.loaded", "number of loaded classes", "1")
+            .assertTypedCounter(
+                "jvm.gc.collections.count",
+                "total number of collections that have occurred",
+                "1",
+                gcCollectionLabels)
+            .register(
+                "jvm.gc.collections.elapsed",
+                metric ->
+                    metric
+                        .hasDescription(
+                            "the approximate accumulated collection elapsed time in milliseconds")
+                        .hasUnit("ms")
+                        .isCounter()
+                        .hasTypedDataPoints(gcCollectionLabels))
```

Comment:
[for reviewer] only a single one has been inlined here, we can do others as a follow-up PR to keep the changes small. For this test suite the test will become a bit more verbose but easier to read.